### ISSUE TITLE
[TW-1573] on-modify hooks can't modify data anymore

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#1599]
_François Charlier on 2015-03-12T17:23:49Z says:_ 

 Commit 1cfdfbae522519ca27216db3bb47ae696da5817a introduced a regression in onModify by not updating back the modified input task after it has passed throught the on-modify hooks.

The old "Transfer the modified task lines back to the original task list" code has been removed but not replaced with code using the new parameters/variables: 
{code:title=src/Hooks.cpp / void Hooks::onModify (const Task& before, Task& after)}
       if (status)
+      {
+        // TODO Hooks debug info needed.
         throw 0;  // This is how hooks silently terminate processing.
+      }
     }
* 
*     // Transfer the modified task lines back to the original task list.
*     tasks.clear ();
*     std::vector <std::string>::iterator i;
*     for (i = input.begin (); i != input.end (); ++i)
*       if (i != input.begin ())
*         tasks.push_back (Task (*i));
   }
{code}

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2015-03-12T17:23:49Z
Modified: 2015-03-12T21:19:56Z
```

**Comment by taskwarrior** [comment]
_François Charlier on 2015-03-12T18:16:34Z says:_ 

 The attached patch fixes the issue

**Comment by taskwarrior** [comment]
_Wim Schuermann on 2015-03-12T21:19:56Z says:_ 

 Thank you for reporting this François, and thank you for the patch.

This has already been fixed in commit [138360b7bc77c838aa62622ffde09ce2d027777a|https://git.tasktools.org/projects/TM/repos/task/commits/138360b7bc77c838aa62622ffde09ce2d027777a] in the 2.4.2 branch. 2.4.2 will probably be released in the next few days, mainly to get this fix out of the door.

I guess I should have opened a "documentation bug" to save you and possibly others some time when running into this - I know I spent a few minutes utterly perplexed why my hook wasn't working as intended :)
